src/types.ts

```diff
@@ -82,7 +87,7 @@ export type GetLibraryManagedProps<C> = JSX.LibraryManagedAttributes<
 export type ConnectedComponent<
   C extends ComponentType<any>,
   P
-> = ComponentType<P> &
+> = FunctionComponent<P> &
```

Comment:
`ComponentType` is a union - your `connect` returns a function component so there was no need to use the `ComponentType` here. Somehow because of that union, the inferred props through `ComponentProps` were also computed as a union, and that caused a problem. This only occurred with older `@types/react` **because** their `FunctionComponent` included `PropsWithChildren`.

I've wanted to write a type test for this - but to pull it off you would have to extend your test suite with a `@types/react` matrix.